alloc-customization-point.html

```diff
@@ -165,11 +180,106 @@ <h2>About the customization points, request for feedback/decision</h2>
   type. In the second approach, it needs only uses_allocator. Types
   like pmr::vector don't care, but types that change their internal
   storage and API based on this decision do care.
+</p>  
+  
+  
+</p>  On further consideration, we concluded that, if a type can convert from a selected allocator
+  to a domain specific allocator, and does so by either specialising the __domain_allocator_traits trait 
+  or providing the domain_alloc_convert function, then the intentions is to have this type behave as if it 
+  "uses_allocator" and therefore the definition of <tt>uses_allocator</tt> should take such a conversion into account. 
+  This allows such a type to get all the benefits of being uses_alloctor, not just the uses_alloctor construction.
+
 </p>
+
 <p>
   LEWG should decide whether uses_allocator should depend on
   __domain_allocator_traits. 
 </p>
 
+<h2>Proposed wording</h2>
+
+
+<p>
+Insert new section after [allocator.uses.trait]:
+</p>
+<blockquote>
+<ins> <h3>Class template domain_allocator_traits</h3>
+[1] <code>namespace std {
+         template &lt;class T, class Alloc&gt; struct domain_allocator_traits {};
+         }
+         </code>
+</p>
+<p>[2] <tt>domain_allocator_traits</tt> provides means to get uses-allocator construction with allocator types other than the
+type specific allocator. Meets the <i>Cpp17BinaryTypeTrait</i> requirements ([meta.rqmts]). </p>
+
+<p>[3]  If <tt>domain_allocator_traits&lt;class T, class Alloc&gt;::value</tt> is <tt>true</tt>, static member
+<tt>domain_allocator_traits&lt;class T, class Alloc&gt;::convert</tt> shall be a function object. 
+
+ </p>
+
+<p>[4] Implementation shall provide a definition of <tt>domain_allocator_traits&lt;class T, class Alloc&gt</tt> that is 
```

Comment:
This seems like we should, in case the implementation provides such a definition, require that that definition would also define a ::convert that calls domain_alloc_convert.

Reply:
How about "If the implementation provides a definition that is derived from true_­type, domain_allocator_traits<;class T, class Alloc>;::convert will be bound to T::domain_alloc_convert."
Will or shall ? 
Is bound a good verb ?

Reply:
"domain_allocator_traits<;class T, class Alloc>::convert is defined as a static member function object that invokes T::domain_alloc_convert", perhaps.

Reply:
i like it, done